src/comparator/wrappers.py

```diff
@@ -778,6 +764,82 @@ def _get_api_version(
         package = self.file_set_pb.file[0].package
         return re.search(version, package).group()
 
+    def _get_source_code_locations_map(
+        self, file_set: descriptor_pb2.FileDescriptorSet
+    ) -> Dict[str, Dict[Tuple[int, ...], descriptor_pb2.SourceCodeInfo.Location]]:
+        source_code_locations_map = {}
+        for fd in file_set.file:
+            # Iterate over the source_code_info and place it into a dictionary.
+            #
+            # The comments in protocol buffers are sorted by a concept called
+            # the "path", which is a sequence of integers described in more
+            # detail below; this code simply shifts from a list to a dict,
+            # with tuples of paths as the dictionary keys.
+            source_code_locations: Dict[
+                Tuple[int, ...], descriptor_pb2.SourceCodeInfo.Location
+            ] = {}
+            for location in fd.source_code_info.location:
+                source_code_locations[tuple(location.path)] = location
+            source_code_locations_map[fd.name] = source_code_locations
```

Comment:
Sure, the dict comprehension looks great to me, and it might be cleaner to create the `source_code_locations` dict first and put it into the map after. Thanks